### PR TITLE
Server error

### LINKED ISSUE #106: Handle server connection errors
## Steps to reproduce the problem

1. remove the API key from `.env.local`
2. start the application

## Expected result after following these steps

Friendly error message explaining that the connection to the server has failed and that they can email tech@extinctionrebellion.nl for help.

If the application is in development mode, it should also suggest to the user reading the `README.md` and properly setting up the server API key.

These error messages should also appear if the user directly goes to the view of an individual role.

## Actual result

Loading... Loading... Loading...

### PR DESCRIPTION
**Pull request by aleesaan**
## What changes have been made?
Added a text to show to the user when there was an error fetching roles.

### Issues resolved by these changes

Resolves #106 

## Additional comments
We should also add error tracking, to notify us if these errors happen on production, but it will be done on a different case.

### CODE REVIEW

**Review by SamLubbers — CHANGES_REQUESTED** [review]
@aleesaan thanks for picking up this issue :raised_hands: 

I would slightly change the implementation so that it handles the server error no matter what page the user visits. Right now the user sees the error message when visiting the `RolesOverview`, but not when they go to a different page, such as viewing an individual role:

image

I think the easiest way to deal with this error is to add the error message in `App.vue`: if the connection to the server fails show a page with only the error message, else, render all the components of the application everything that is in `v-app` at the moment. I would show this error message big and centred, something like this :stuck_out_tongue_winking_eye: 

image
 
I would also reuse the support contact details in `TheAppBar`. That way if we change these contact details it automatically changes in all places. The easiest way to do this is to make a shared component used in both places. As you can see in `TheApprBar`, we still use the integration@extinctionrebellion.nl email. You can change that with tech@extinctionrebellion.nl

As a last small detail, I would make the text `set the server API key` a hyperlink to: https://github.com/xrnl/aginun#install That way developers know how they can set the API key.

File: src/views/ErrorPage.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      </a>
+    </p>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "ErrorPage",
+  data: () => ({
+    isDevMode: process.env.NODE_ENV === "production",
```
Comment (SamLubbers) [inline src/views/ErrorPage.vue:36]: small typo here :wink: 
```suggestion
    isDevMode: process.env.NODE_ENV !== "production",
```

**Review by SamLubbers — CHANGES_REQUESTED** [review]
Hey @aleesaan thanks for implementing the feedback :smile: This error page is already so much better than the previous version, as the user cannot interact with the application when the connection to the server fails.

### More resilient error handling

I think, however, that we can make the application even more robust. Right now the error page shows whenever the user goes to one of the roles pages (`/roles`) and the application fails to load roles. This works fine for now, but once we add more pages (`/tasks`, `/settings`) the error page will not load even if the connection with the server fails. That is why I suggest adding the error handling within a [beforeCreate lifecycle hook](https://vuejs.org/v2/api/#beforeCreate) in `App.vue`. That way, if the connection to the server fails, the application will not even attempt to load the roles.

In summary, the current flow is as follows:
  1. Application loads
  2. Application tries to load roles
  3. There is an error connecting to the server --> error page is shown

The new flow would be as follows:
  1. Application loads
  2. Inside the `beforeCreate` lifecycle hook of App.vue, the application tries to ping the server to see if a connection can be established.
  3. There is an error connecting to the server --> error page is shown

### Shared contact details

Right now the email `tech@extinctionrebellion.nl` is hardcoded into `TheAppBar.vue` and `ErrorPage.vue`. I suggest exporting the email into a variable (`contactEmail`), so that if this email changes it is automatically updated in all parts of the application. 

So you could create a file with all application constants:

```js
export const contactEmail = 'tech@extinctionrebellion.nl'
```
and then import this email in all places that you use it

```html
<a :href=`mailto:${contactEmail}`>
   {{ contactEmail }}
</a>
```

File: src/views/ErrorPage.vue (hunk 1 above)
Comment (aleesaan) [reply src/views/ErrorPage.vue:36]: ahah thanks!

**Comment by aleesaan** [comment]
Your suggestion sounds good to me! But wouldn't we need to also to configure an endpoint in the backend so that it can be pinged? Using one of the existent queries would be kind of random...

**Comment by SamLubbers** [comment]
@aleesaan I had assumed that Hasura would have a default endpoint to ping. If it doesn't have it then we can indeed set up a custom one :ok_hand:

**Review by SamLubbers — APPROVED** [review]
@aleesaan this is PERFECT :star_struck: now this functionality is even better as the error page loads directly (instead of the roles page loading for a fraction of a second and then the error page showing). Also this saves us a lot of code checking for server errors in other parts of the application.

**Pull request merged**